control/trapezoid_velocity_profile.go

```diff
@@ -122,6 +130,10 @@ func (s *trapezoidVelocityGenerator) reset() error {
 	if s.kppGain == 0 {
 		s.kppGain = 0.45
 	}
+	s.lastVelCmd = 0
+	if s.y != nil {
+		s.lastVelCmd = s.y[0].GetSignalValueAt(0)
+	}
```

Comment:
I'm not clear on what this is doing

Reply:
if the trapezoid block already has a desired velocity signal, we should preserve that state in `lastVelCmd`. this was required to persist velocity commands thru multiple `SetRPM` calls. It's possible the equations should be tweaked so that we no longer assume a non-zero velocity in the math, but I think this works for now.